src/esp/gfx/replay/Player.h

```diff
@@ -101,10 +107,14 @@ class Player {
    */
   const std::vector<Keyframe>& debugGetKeyframes() const { return keyframes_; }
 
+  /**
+   * @brief Applies a keyframe to the player.
+   */
+  void applyKeyframe(const Keyframe& keyframe);
```

Comment:
I'm flexible on all the JS code in this PR, but for the Player class, we need to rethink your API changes.

It's dangerous to give users the freedom to manage and apply their own keyframes, because keyframes are "incremental"--they generally only store the changes since the previous frame.

Reply:
That makes sense, I have changed it to better demonstrate this "incremental" idea.